### DISCUSSION TITLE
Confused where the session in the provider is coming

### DISCUSSION THREAD

**Discussion by mirabledictu** [nextauthjs/next-auth#2530] [Help] [answered]
Hello, just want to ask something because I'm confused. Pardon me. 

In the examples, like https://github.com/nextauthjs/next-auth-example, the `_app.js` file includes the provider and has the session passed like this:

```js
import { Provider } from "next-auth/client"
import type { AppProps } from "next/app"

export default function App({ Component, pageProps }: AppProps) {
  return (
    <Provider
      session={pageProps.session}
    >
      <Component {...pageProps} />
    </Provider>
  )
}
```

How does `pageProps.session` gets the session? Since the examples doesn't have a `getServerSideProps` except the `/server` page.. or am I missing something?

**Comment by balazsorban44** [comment, marked as answer]
no, you aren't. it's a tradeoff. we can't magically make the session appear so it has to come from getInitialProps or getServerSideProps. if it's not passed, we fetch it from the client at initialization, but you will have to accommodate for the initial load


Ref: https://next-auth.js.org/getting-started/client#provider
